torch/csrc/jit/codegen/cuda/scheduler/utils.cpp

```diff
@@ -805,7 +808,9 @@ PersistentBufferSizeReturn persistentBufferSize(
     persistent_buffer_sizes[buffer_i] = persistent_buffer_sizes[buffer_i] == -1
         ? 0
         : persistent_buffer_sizes[buffer_i] *
-            dataTypeSize(buffer->getDataType().value());
+            dataTypeSize(
+                buffer->getDataType().value(),
+                indexModeToDtype(runtime_info.getIndexMode()));
```

Comment:
Can a buffer type be the Index type? Does it happen with a Welford op?

Reply:
Yes, welford's output is of index type.